### PR TITLE
just worked on messaging ui template

### PR DESCRIPTION
**Pull request by MUGABA**
### What this PR does (required):
- This pr contains code for the messaging interface

### Screenshots / Videos (front-end only):
- Post at least one screenshot to show the feature in action. **Only required in front-end work**

Screenshot 2022-02-16 124900

### CODE REVIEW

File: client/src/components/Messaging/ActiveChatMember/ActiveChatMember.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import { useStyles } from './useStyles';
+
+const ActiveChatMember = (): JSX.Element => {
+  const classes = useStyles();
+  return (
+    <Box className={classes.cardContainer}>
+      <Box className={classes.image}>
+        <img src="https://cdn.pixabay.com/photo/2021/09/12/18/07/robin-6619184_960_720.jpg" alt="Image" />
+        <p className={classes.online}></p>
```
Comment (ezeilo-su) [inline client/src/components/Messaging/ActiveChatMember/ActiveChatMember.tsx:12]: Let's make use of MUI `Typography` component here

File: client/src/components/Messaging/ChatSection/ChatRoom.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+  };
+
+  return (
+    <Box>
+      <ActiveChatMember />
+      <Box className={classes.chatRoom}>
+        {messages &&
+          messages.map((message, i) => <ChatMessage key={i} sender={message.sender} message={message.message} />)}
```
Comment (ezeilo-su) [inline client/src/components/Messaging/ChatSection/ChatRoom.tsx:48]: Let's avoid using indexes as keys as this is an anti-pattern in React.

File: client/src/components/Messaging/Message/ChartMessage.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+}
+const ChatMessage = ({ sender, message }: props): JSX.Element => {
+  const classes = useStyles();
+  const messageClass = sender === 'me' ? classes.sent : classes.received;
+
+  const textStyle = messageClass.split('-')[1] === 'sent' ? classes.textSent : classes.textRecieved;
+
+  return (
+    <>
+      <div className={`message ${messageClass}`}>
```
Comment (ezeilo-su) [inline client/src/components/Messaging/Message/ChartMessage.tsx:16]: Let's use MUI `Box` component to replace this `div`

File: client/src/components/Messaging/Message/ChartMessage.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        <img src="https://cdn.pixabay.com/photo/2021/09/12/18/07/robin-6619184_960_720.jpg" />
```
Comment (ezeilo-su) [inline client/src/components/Messaging/Message/ChartMessage.tsx:17]: This element is missing an `alt` attribute.

File: client/src/components/Messaging/SenderCard/MessageSenderCars.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+import { useStyles } from './useStyles';
+
+const MessageSenderCard = (): JSX.Element => {
+  const classes = useStyles();
+  return (
+    <Box className={classes.cardContainer}>
+      <Box className={classes.image}>
+        <img src="https://cdn.pixabay.com/photo/2021/09/12/18/07/robin-6619184_960_720.jpg" alt="Image" />
+        <p className={classes.online}></p>
```
Comment (ezeilo-su) [inline client/src/components/Messaging/SenderCard/MessageSenderCars.tsx:11]: Replace MUI component

File: client/src/components/Messaging/SenderCard/MessageSenderCars.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { Typography, Box } from '@mui/material';
+
+import { useStyles } from './useStyles';
+
+const MessageSenderCard = (): JSX.Element => {
+  const classes = useStyles();
+  return (
+    <Box className={classes.cardContainer}>
+      <Box className={classes.image}>
+        <img src="https://cdn.pixabay.com/photo/2021/09/12/18/07/robin-6619184_960_720.jpg" alt="Image" />
```
Comment (ezeilo-su) [inline client/src/components/Messaging/SenderCard/MessageSenderCars.tsx:10]: Add an `alt` attribute

File: client/src/components/Messaging/SendersWrapper/useStyles.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  wrapper: {
+    width: '420px',
+    boxShadow: '4px 4px 13px 7px rgba(217,217,217,0.26)',
+    padding: theme.spacing(2),
+    background: 'white',
+  },
+  header: {
+    width: '400px',
+    height: '100px',
+    // boxShadow: '4px 4px 13px 7px rgba(217,217,217,0.26)',
```
Comment (ezeilo-su) [inline client/src/components/Messaging/SendersWrapper/useStyles.ts:14]: Remove comment

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Good work with this one. Please have a look at those comments and ensure to format your code properly.

File: client/src/components/Messaging/Message/ChartMessage.tsx (hunk 4 above)
Comment (moffatethan) [reply client/src/components/Messaging/Message/ChartMessage.tsx:17]: Could we address this comment?

File: client/src/components/Messaging/Message/ChartMessage.tsx (hunk 3 above)
Comment (moffatethan) [reply client/src/components/Messaging/Message/ChartMessage.tsx:16]: Please address this comment Rashid 🙂

File: client/src/components/Messaging/SenderCard/MessageSenderCars.tsx (hunk 5 above)
Comment (moffatethan) [reply client/src/components/Messaging/SenderCard/MessageSenderCars.tsx:11]: Please use the Typography component

File: client/src/components/Messaging/SendersWrapper/useStyles.ts (hunk 7 above)
Comment (moffatethan) [reply client/src/components/Messaging/SendersWrapper/useStyles.ts:14]: Remove this comment please

File: server/utils/socket.js [hunk 8]
```diff
@@ -8,9 +8,11 @@ module.exports = async (server, app) => {
...
-    const decoded = jwt.verify(token, process.env.JWT_SECRET);
+    if (token) {
+      const decoded = jwt.verify(token, process.env.JWT_SECRET);
 
-    currentUser = await User.findById(decoded.id);
+      currentUser = await User.findById(decoded.id);
+    }
 
     next();
   });
```
Comment (moffatethan) [inline server/utils/socket.js:18]: Does this still establish connection to our socket if they are not authenticated? Could you confirm for me 🙂

**Review by moffatethan — CHANGES_REQUESTED** [review]
Nice work - just some comments to address

**Comment by MUGABA** [comment]
All comments are resolved

File: client/src/components/Messaging/Message/ChartMessage.tsx (hunk 4 above)
Comment (MUGABA) [reply client/src/components/Messaging/Message/ChartMessage.tsx:17]: It was addressed sir

**Review by moffatethan — APPROVED** [review]
Address merge conflict then you can merge it in

**Pull request merged**